src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -309,9 +318,33 @@ private RecurringDisbursementVoucherPDPStatus buildRecurringDisbursementVoucherP
         pdpStatus.setPdpStatus(disbursementVoucherDocument.getDisbursementVoucherPdpStatus());
         pdpStatus.setPaymentDetailDocumentType(disbursementVoucherDocument.getPaymentDetailDocumentType());
         pdpStatus.setDueDate(disbursementVoucherDocument.getDisbursementVoucherDueDate());
+        pdpStatus.setDvStatus(disbursementVoucherDocument.getDocumentHeader().getWorkflowDocument().getStatus().getLabel());
         return pdpStatus;
     }
-
+    
+    @Override
+    public Set<String> cancelPDPPayments(RecurringDisbursementVoucherDocument recurringDisbursementVoucherDocument, String cancelMessage) {
+        Collection<PaymentDetail> paymentDetails = findPaymentDetailsFromRecurringDisbursementVoucher(recurringDisbursementVoucherDocument);
+        Set<String> canceledPaymentGroups = new HashSet<String>();
+        for (PaymentDetail detail : paymentDetails) {
+            String paymentDetailStatus = detail.getPaymentGroup().getPaymentStatusCode();
+            if (isPaymentCancelableByLoggedInUser(paymentDetailStatus)) {
+                LOG.debug("cancelPDPPayments: About to Cancel " + detail.getId() + " for a reason of " + cancelMessage);
+                boolean canceledSuccessfully = getRecurringDisbursementVoucherPaymentMaintenanceService().cancelPendingPayment(detail.getPaymentGroupId().intValue(), 
+                        detail.getId().intValue(), cancelMessage, GlobalVariables.getUserSession().getPerson());
```

Comment:
Since GlobalVariables.getUserSession().getPerson() will not change during this for loop might be better to get this into a local variable before the loop starts